File: packages/ios/Sources/Components/Progress/Progress.figma.swift
import SwiftUI
import Figma

struct KozmosProgressConnect: FigmaConnect {
    let component = KozmosProgress.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=83-252"

    @FigmaEnum(
        "Value",
        mapping: [
            "0": "0",
            "25": "25",
            "50": "50",
            "75": "75",
            "100": "100"
        ]
    )
    var value: String = "0"

    var body: some View {
        KozmosProgress(value: self.progressValue)
    }

    private var progressValue: Double {
        (Double(self.value) ?? 0) / 100
    }
}

File: packages/android/src/main/java/com/kozmos/components/Progress/Progress.figma.kt
package com.kozmos.components.progress

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=83-252")
class KozmosProgressConnect {
    @FigmaProperty(FigmaType.Enum, "Value")
    val value: String = Figma.mapping(
        "0" to "0",
        "25" to "25",
        "50" to "50",
        "75" to "75",
        "100" to "100"
    )

    @Composable
    fun ComponentExample() {
        KozmosProgress(progress = (value.toFloatOrNull() ?: 0f) / 100f)
    }
}
